core/spec.md

```diff
@@ -1754,16 +1988,18 @@ Regardless of how the model is retrieved, the overall format is as follows:
 ```
 
 The following describes the attributes of Registry model:
-- `schemas`
-  - A list of schema formats in which the Registry model can be returned. Each
-    value MUST be a schema document format name (e.g. `jsonSchema/2020-12`),
-    and SHOULD be of the form `NAME[/VERSION]`. All implementations of this
-    specification MUST support `xRegistry-json` (the JSON serialization as
-    defined by this specification), and SHOULD include it in the list.
-  - Type: String.
-  - OPTIONAL.
-  - MUST be case insensitive.
-  - MUST be a read-only attribute in API view.
+- `labels`
+  - A set of name/value pairs that allows for additional metadata about the
+    Registry to be stored without changing the schema of the model.
+  - If present, MUST be a map of zero or more name/value string pairs.
+  - Keys MUST be non-empty strings.
+  - Values MAY be empty strings.
+  - The following labels are defined for convenience and MAY also be used
+    as labels for Groups or Resource model types:
+    - Name: `compatiblewith`
+      Value: A URI reference to Registry that this Registry's model is
+      compatible with. This specification does not mandate any runtime
+      validation of this claim.
```

Comment:
Here we need to clarify how to point to a particular model, it's not sufficient to point to an xregistry itself as it could contain all kinds of models. Compatibility will always be between particular models, not between the registries as a whole.

```suggestion
    - Name: `compatiblewith`
      Value: A URI reference to an xRegistry model that this model is
      compatible with. All models that share the same reference are interoperable. 
      This specification does not mandate any runtime validation of this claim.
```

But we need to clarify probably a few more points:
* How to construct this URL? Ideally, we would point to the model, e.g. on group side we'd point from the group model to e.g. https://xregistry.cncf.org/model#groups/schemagroups (do we have a link where I can retrieve just a partial model? Otherwise we need anchor tags here. 
* What does compatible mean? 
  * It's primarily about ecosystem compatibility around a model (like we have with OpenAPI, JSON Schema of a "major" version). 
  * There's import and export compatibility, which depends on how the own model was extended:
    * Adding optional attributes: Full import and export compatibility
    * Removing optional attributes: Full import and export compatibility
    * Adding further constraints to existing attributes: Full import and export compatibility
    * Adding new mandatory attribute or making optional attribute mandatory: Full export compatibility, not import.

This is probably too much for the place here? Where would you put content that explains a concept in more detail, to not overload the spec itself?

One open question: Should we also point out a version of the model here? Most specs at least have a <major>.<minor> version, although the compatibility should be sufficiently expressed with just <major>. Maybe a second label? New minor releases can indicate additional attributes / features in the model.

Reply:
I suspect we may want to treat these "well-known labels" the same way we treat extensions in CloudEvents. Meaning, anyone can propose new ones at any time and they might even change their definitions over time as people start to use them (or even remove them). So, it might be best to create a 2nd doc that has a different lifecycle than the spec itself and has a much lower bar for edits (and no official versioning). WDYT?

Reply:
Yes, I like the idea! 

We also need to put the "compatiblewith" into the 3 CNCF provided models, so when you import them you already have them by default. That would make this label also more visible. 

I thought of another simplification: How about you can just add any URI, and each model defines which it is? You could also point to the model specification, which may be more user friendly e.g. https://github.com/xregistry/spec/blob/main/endpoint/spec.md

Reply:
@duglin I can see three kinds of well-known labels: 1) labels on the meta-model (like `compatiblewith` and `modelversion` (?)), 2) labels on the instance level (not sure we have any) and 3) labels defined as part of a model, e.g. the endpoint model.

Here is a PR to add it to the extension specs: https://github.com/xregistry/spec/pull/225